### DISCUSSION TITLE
Should provider_groups be single object rather than array

### DISCUSSION THREAD

**Discussion by theogaraj** [CMSgov/price-transparency-guide#649] [Q&A]
The [schema specific to the provider reference object](https://github.com/CMSgov/price-transparency-guide/tree/master/schemas/in-network-rates#provider-reference-object) defines `provider_groups` to be an array of provider objects.  However, a couple of things make me wonder if that's incorrect and it should actually just be a single `provider_group` object:

* the description for `provider_group_id` (the previous element of the provider reference object) is described as "The unique, primary key for the associated provider_group object", which suggests a single key for a single provider group object.  Conceptually it makes more sense to have a single ID for a single group rather than a number of groups.
* the [referenced  example](https://github.com/CMSgov/price-transparency-guide/blob/c3ba257f41f4b289b574557e2fcf0833c36ef79f/examples/in-network-rates/in-network-rates-fee-for-service-single-plan-sample.json#L10-L28) has a single `provider_group` object.

I haven't seen any discussions on this but wonder if anybody's got any thoughts?

**Comment by shaselton-usds** [comment]
@theogaraj The "referenced example" appears to be an older commit that was updated by [this commit](https://github.com/CMSgov/price-transparency-guide/blob/2df1618b6fe29f12cdcf55e81e5707da29c6608c/examples/in-network-rates/in-network-rates-fee-for-service-single-plan-sample.json#L12-L24) to reflect the correction.

It is an array because there may be negotiated arrangements across multiple provider groups that are exactly the same so the need to define each group separately can be avoided and also the overhead of managing multiple `provider_group_id`s for the same negotiated rate.

Definitely open to additional thoughts though!

**Reply by theogaraj** [reply]
@shaselton-usds  but `provider_references` within the `negotiated_rate_details` object ([link](https://github.com/CMSgov/price-transparency-guide/tree/master/schemas/in-network-rates#negotiated-rate-details-object)) is already an array.  So if a payer has the same negotiated arrangement with multiple provider groups, it:
- defines each group once in the main `provider_references` section of the file, with _one group per group ID_.
- defines the negotiated arrangement _once_, and then specifies each of the relevant group ids within the `provider_references` array of that negotiated rate object.

Consider a scenario where:
-  `Procedure1`:  payer has _identical_ negotiated arrangements with both groups.
-  `Procedure2`: payer has a _different_ negotiated arrangement with each group.

In the example you linked to (updated commit), there are _two_ provider groups (ein 11-1111111 and ein 22-2222222) which both have the _same_ `provider_group_id` of '1'.  By having a 1:many relationship between group IDs and the groups themselves, the schema doesn't allow the payer to distinguish between the two groups to specify different negotiated rates for `Procedure2`.  They would have to explicitly define the entire group within the negotiated rate object (less efficient than simply using the reference).

Instead, if we have a 1:1 relationship between _group id_ and _group_ as below (note single `provider_group` below rather than array of `provider_groups`) ...

```
"provider_references":[
    {
         "provider_group_id": 1,
         "provider_group": {
              "npi": [1111111111, 2222222222, 3333333333, 4444444444, 5555555555],
              "tin":{
                "type": "ein",
                "value": "11-1111111"
              }
    },
   {
         "provider_group_id": 2
         "provider_group":{
               "npi": [1111111111, 2222222222, 3333333333, 4444444444, 5555555555],
               "tin":{
                 "type": "ein",
                 "value": "22-2222222"
                  }
    }
]
```

... then in the `in_network` section of the file we could support the single arrangement with both groups for `Procedure1` (note both group id 1 and 2 in the single negotiated rate object for `Procedure1`, as well as  different negotiated arrangements for `Procedure2` (note the two different negotiated rate objects, one per group).
```
"in_network": [
    {
       ...
       "billing_code": "Procedure1",
       "negotiated_rates": [
           {
             "provider_references": [1, 2],
             "negotiated_prices": [
                ...
              ]
           }
        ]
    },
   {
      ...
      "billing_code": "Procedure2",
      "negotiated_rates": [
         {
            "provider_references": [1],
            "negotiated_prices": [
                ...
            ]
         },
        {
            "provider_references": [2],
            "negotiated_prices": [
                ...
            ]
         }
]
```
